packages/types/src/infrastructure/httpServer/BaseController.ts

```diff
@@ -62,19 +62,15 @@ export abstract class BaseController {
             .send(buffer);
     }
 
-    protected async qrCode<T extends Result<any>>(
-        result: T,
-        qrPredicate: (result: T) => Promise<QRCode>,
-        filename: string,
-        response: express.Response,
-        status: number
-    ): Promise<void> {
+    protected async qrCode<T extends Result<{ reference: { url: string } }>>(result: T, filename: string, response: express.Response, status: number): Promise<void> {
```

Comment:
You might as well only pass the url.

```suggestion
    protected async qrCode(url: string, filename: string, response: express.Response, status: number): Promise<void> {
```

Reply:
As written per Teams, not possible as this function should call guard on the Result.